azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Certificate.java

```diff
@@ -0,0 +1,594 @@
+      }
+   }
+
+   @AutoValue
+   public abstract static class AdministrationDetails {
+      @Nullable
+      public abstract String email();
+
+      @Nullable
+      public abstract String firstName();
+
+      @Nullable
+      public abstract String lastName();
+
+      @Nullable
+      public abstract String phoneNumber();
+
+      @SerializedNames({"email", "first_name", "last_name", "phone"})
+      public static AdministrationDetails create(final String email,
+                                                 final String firstName,
+                                                 final String lastName,
+                                                 final String phoneNumber) {
+         return new AutoValue_Certificate_AdministrationDetails(email, firstName, lastName, phoneNumber);
+      }
+   }
+
+   @AutoValue
+   public abstract static class IssuerBundle {
+      @Nullable
+      public abstract IssuerAttributes attributes();
+
+      @Nullable
+      public abstract IssuerCredentials credentials();
+
+      @Nullable
+      public abstract String id();
+
+      @Nullable
+      public abstract OrganizationDetails organizationDetails();
+
+      @Nullable
+      public abstract String provider();
+
+      @SerializedNames({"attributes", "credentials", "id", "org_details", "provider"})
+      public static IssuerBundle create(final IssuerAttributes attributes,
+                                        final IssuerCredentials credentials,
+                                        final String id,
+                                        final OrganizationDetails orgDetails,
+                                        final String provider) {
+         return new AutoValue_Certificate_IssuerBundle(attributes, credentials, id, orgDetails, provider);
+      }
+   }
+
+   @AutoValue
+   public abstract static class Contact {
+      @Nullable
+      public abstract String email();
+
+      @Nullable
+      public abstract String name();
+
+      @Nullable
+      public abstract String phone();
+
+      @SerializedNames({"email", "name", "phone"})
+      public static Contact create(final String email,
+                                   final String name,
+                                   final String phone) {
+         return new AutoValue_Certificate_Contact(email, name, phone);
+      }
+   }
+
+   @AutoValue
+   public abstract static class Contacts {
+      @Nullable
+      public abstract List<Contact> contacts();
+
+      @Nullable
+      public abstract String id();
+
+      @SerializedNames({"contacts", "id"})
+      public static Contacts create(final List<Contact> contacts,
+                                    final String id) {
+         return new AutoValue_Certificate_Contacts(contacts, id);
+      }
+   }
+
+   @AutoValue
+   public abstract static class DeletedCertificateBundle {
+      @Nullable
+      public abstract CertificateAttributes attributes();
+
+      @Nullable
+      public abstract String bytes();
```

Comment:
[minor] The method name suggests a return type of `byte[]`, but let's keep it as-is if a string really makes more sense given the content of the field.